src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/DirectoryControl.cs

```diff
@@ -389,7 +472,19 @@ public SearchOption SearchOption
 
         public override byte[] GetValue()
         {
-            _directoryControlValue = BerConverter.Encode("{i}", new object[] { (int)SearchOption });
+            int sizeEstimate = 8;
+            AsnWriter writer = new(AsnEncodingRules.BER, sizeEstimate);
+
+            /* This is as laid out in MS-ADTS, 3.1.1.3.4.1.12.
+             * SearchOptionsRequestValue ::= SEQUENCE {
+             *                                  Flags     INTEGER }
+             */
+            using (AsnWriter.Scope outerSequence = writer.PushSequence())
```

Comment:
We used a nested type here because the expected code is just

```suggestion
            using (writer.PushSequence())
```